### PR TITLE
Make timer clock soure configurable

### PR DESCRIPTION
**Pull request by marcfir**
Hi Robert,
here is a suggestion for configuring the clock source.

### CODE REVIEW

File: src/posix/timer.c [hunk 1]
```diff
@@ -17,135 +17,142 @@
...
-        if (ret == 0) {
-            break;
-        }
-
-        ts = rest;
+  struct timespec ts = {(nsec / NSEC_PER_SEC), (nsec % NSEC_PER_SEC)};
+  struct timespec rest;
+
+  while (1) {
+    int ret = clock_nanosleep(LIBOSAL_CLOCK, 0, &ts, &rest);
```
Comment (robert-burger) [inline src/posix/timer.c]: should also be global_clock_id instead of LIBOSAL_CLOCK

**Review by robert-burger — CHANGES_REQUESTED** [review]
1. this is meant to use CLOCK_MONOTIC, CLOCK_REALTIME, ... directly from system headers. should we add some defines to that os-specific values like e.g.  LIBOSAL_CLOCK_MONOTONIC?
2. can you please fix indenting back to 4 spaces? changes are also a little hard to read now...

**Comment by marcfir** [comment]
1. I thought about this. Linux has more clock types than the POSIX standard, so how would you expose the clock ID's? Just the ones from POSIX and if you want to use an OS specific one you can include the time.h header? 

2. Oh yeah, I will fix that.

3. I added a little test locally that configures different times. The test is not really deterministic because getting two times and comparing the difference to an upper bound could be interrupted by the scheduler. But I had no idea how to make the test more deterministic. I can add the test if you like.

**Comment by marcfir** [comment]
What to do with the `pthread_condattr_setclock()` calls?

**Comment by robert-burger** [comment]
> What to do with the `pthread_condattr_setclock()` calls?

couldn't we do e.g. osal_timer_get_clock_source ? and return the internal clock id integer number?

**Comment by marcfir** [comment]
> > What to do with the `pthread_condattr_setclock()` calls?
> 
> couldn't we do e.g. osal_timer_get_clock_source ? and return the internal clock id integer number?

Good point. I added the call. It should work.

**Review by robert-burger — APPROVED** [review]
lgtm

**Comment by robert-burger** [comment]
thanks

**Pull request merged**